pkg/credentialprovider/azure_credentials.go

```diff
@@ -224,10 +225,16 @@ func (a *acrProvider) getFromACR(ctx context.Context, loginServer string) (strin
 // If the provided image is not an acr image, this function will return an empty string.
 func (a *acrProvider) parseACRLoginServerFromImage(image string) (string, string) {
 	targetImage, sourceRegistry := a.processImageWithRegistryMirror(image)
-
-	match := acrRE.FindAllString(targetImage, -1)
-	if len(match) == 1 {
-		targetRegistry := match[0]
+	targetRegistry := acrRE.FindString(targetImage)
+	imageWithoutRegistry := strings.Trim(targetImage, targetRegistry)
```

Comment:
is `strings.Trim` here expected? https://pkg.go.dev/strings#Trim It's to remove every character in the 2nd parameter from the leading/trailing of the first parameter

Reply:
The aim is to remove the leading targetRegistry from targetImage, i.e. to remove acr from the image.
Updated the PR to use TrimPrefix to be specific.